### PR TITLE
Test stabilizations

### PR DESCRIPTION
**Pull request by sutaakar**

### CODE REVIEW

File: framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    public KieServerMgmtControllerClient getKieServerMgmtControllerClient() {
+        return kieServerMgmtControllerClient;
+    }
+
+    /**
+     * Wait until any server template is created in controller.
+     */
+    public void waitForServerTemplateCreation() {
+        long timeoutTime = Calendar.getInstance().getTimeInMillis() + 30000L;
```
Comment (jiripetrlik) [inline framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java]: Would it be possible to use new Date time API in Java 8?

Comment (sutaakar) [inline framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java]: Good catch, updated.

File: framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+package org.kie.cloud.common.provider;
+
+import java.time.Instant;
+import java.util.Collection;
+
+import org.kie.cloud.api.deployment.WorkbenchDeployment;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.integrationtests.controller.client.KieServerMgmtControllerClient;
+
+public class KieServerMgmtClientProvider {
```
Comment (jhrcek) [inline framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java]: Why call it KieServerMgmtClientProvider if it provides KieServerMgmtControllerClient?
It'd prefer KieServerControllerClientProvider

File: framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            }
+
+            try {
+                Thread.sleep(200L);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException("Interrupted while waiting for server template creation.", e);
+            }
+        }
+        throw new RuntimeException("Timeout while waiting for server template creation.");
```
Comment (jhrcek) [inline framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java]: I usually include timeout time in the exception message to make it clear without looking at the sources how long was the time (how slow it was).

Like "Timeout while waiting for 30 seconds for server template creation" or such

File: framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java (hunk 2 above)
Comment (jakubschwan) [inline framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java]: Better, but longer description of this class can be KieServerMgmtControllerClientProvider

Comment (sutaakar) [inline framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java]: I think that KieServerControllerClientProvider should be enough, adjusted.

File: framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java (hunk 3 above)
Comment (sutaakar) [inline framework-cloud/framework-cloud-common/src/main/java/org/kie/cloud/common/provider/KieServerMgmtClientProvider.java]: Done.

**Comment by jiripetrlik** [comment]
Looks good for me now.

**Pull request merged**